src/AsmResolver.DotNet/Code/Cil/CilMethodBody.cs

```diff
@@ -13,6 +15,256 @@ namespace AsmResolver.DotNet.Code.Cil
+                }
+
+            // Resolve all operands.
+            foreach (var instruction in result.Instructions)
+                instruction.Operand =
+                    ResolveOperandReflection(result, instruction, operandResolver, tokenList, importer) ??
+                    instruction.Operand;
+            return result;
+        }
+
+        private static object ResolveOperandReflection(CilMethodBody methodBody, CilInstruction instruction,
+            ICilOperandResolver resolver, List<object> Tokens, ReferenceImporter Importer)
+        {
+            switch (instruction.OpCode.OperandType)
+            {
+                case CilOperandType.InlineBrTarget:
+                case CilOperandType.ShortInlineBrTarget:
+                    return new CilInstructionLabel(
+                        methodBody.Instructions.GetByOffset(((ICilLabel) instruction.Operand).Offset));
+
+                case CilOperandType.InlineField:
+                case CilOperandType.InlineMethod:
+                case CilOperandType.InlineSig:
+                case CilOperandType.InlineTok:
+                case CilOperandType.InlineType:
+                    return ReadToken(((MetadataToken) instruction.Operand).ToUInt32(), Tokens, Importer);
+                case CilOperandType.InlineString:
+                    return ReadToken(((MetadataToken) instruction.Operand).ToUInt32(), Tokens, Importer);
+                case CilOperandType.InlineSwitch:
+                    var result = new List<ICilLabel>();
+                    var labels = (IEnumerable<ICilLabel>) instruction.Operand;
+                    foreach (var label in labels)
+                    {
+                        var target = methodBody.Instructions.GetByOffset(label.Offset);
+                        result.Add(target == null ? label : new CilInstructionLabel(target));
+                    }
+
+                    return result;
+
+                case CilOperandType.InlineVar:
+                case CilOperandType.ShortInlineVar:
+                    return resolver.ResolveLocalVariable(Convert.ToInt32(instruction.Operand));
+
+                case CilOperandType.InlineArgument:
+                case CilOperandType.ShortInlineArgument:
+                    return resolver.ResolveParameter(Convert.ToInt32(instruction.Operand));
+
+                case CilOperandType.InlineI:
+                case CilOperandType.InlineI8:
+                case CilOperandType.InlineNone:
+                case CilOperandType.InlineR:
+                case CilOperandType.ShortInlineI:
+                case CilOperandType.ShortInlineR:
+                    return instruction.Operand;
+
+                case CilOperandType.InlinePhi:
+                    throw new NotSupportedException();
+
+                default:
+                    throw new ArgumentOutOfRangeException();
+            }
+        }
+
+        private static object ReadToken(uint token, IList<object> Tokens, ReferenceImporter importer)

```

Comment:
- Define `token` as `MetadataToken` and use `token.Rid` and `token.Table` instead of magic constants and shift operations.
- `Tokens` -> `tokens`